galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java

```diff
@@ -0,0 +1,148 @@
+                throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            }
+        }
+    }
+    
+    public void processGalasaProperty (JsonObject resource, String action) throws InternalServletException{
+        try{
+            String apiversion = resource.get("apiVersion").getAsString();
+            String expectedApiVersion = GalasaProperty.DEFAULTAPIVERSION;
+            if (apiversion.equals(expectedApiVersion)){
+                GalasaProperty galasaProperty = gson.fromJson(resource, GalasaProperty.class);           
+                if (galasaProperty.isPropertyValid()){
+                    CPSFacade cps = new CPSFacade(framework);
+                    CPSNamespace namespace = cps.getNamespace(galasaProperty.getNamespace());
+                    CPSProperty property = namespace.getPropertyFromStore(galasaProperty.getName());
+                    /*
+                    * The logic below is used to determine if the XNOR in property.setPropertyToStore will action the request or error
+                    * Logic Table to Determine actions
+                    * If the action is equal to "update" (force update) the updateProperty is set to true (update property,
+                    * will error if the property does not exist in CPS)
+                    * If the action is either "update" or "apply" and the property exists in CPS the updateProperty is set to true (update property)
+                    * If the action is equal to "apply" and the property does not exist in CPS the updateProperty is set to false (create property)
+                    * If the action is equal to "create" (force create) the updateProperty is set to false (create property, will error if the property exists in CPS)
+                    */
+                    boolean updateProperty = false;
+                    if ((updateActions.contains(action) && property.existsInStore()) || action.equals("update")){
+                        updateProperty = true;
+                    }
+                    property.setPropertyToStore(galasaProperty, updateProperty);
+                }
+            }else{
+                ServletError error = new ServletError(GAL5027_UNSUPPORTED_API_VERSION, apiversion, expectedApiVersion);
+                throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
+            }
+        }catch (InternalServletException i){
+            throw i;
+        }catch (NullPointerException n){
+            // Null Pointer exception caused by bad Key Names in JSON i.e. apiversion instead of apiVersion
+            ServletError error = new ServletError(GAL5400_BAD_REQUEST,resource.toString());
+            throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
+
+        }catch (Exception e){
+        ServletError error = new ServletError(GAL5000_GENERIC_API_ERROR, e.getMessage());
```

Comment:
is indenting messed up here ? Might be a github viewer problem. Use spaces instead of tabs.